### PR TITLE
Simplify complex hybrid energy

### PR DESCRIPTION
**Pull request by bhourahine**
This touches the hybrid routines, but also changes to using a common routine for the time dependent propagation energy evaluation with hybrids.

### CODE REVIEW

**Review by vanderhe — APPROVED** [review]
LGTM

**Comment by vanderhe** [comment]
@bhourahine I will rebase #1457 on top of this one as soon as merged.

**Comment by bhourahine** [comment]
@fbonafe I'll merge this tomorrow unless you have comments.

**Pull request merged**
